### PR TITLE
Updated linked_domain README

### PR DESCRIPTION
**Pull request by orangejenny**
Update to reflect changes made in https://github.com/dimagi/commcare-hq/issues/25718.

### CODE REVIEW

File: corehq/apps/linked_domain/README.md [hunk 1]
```diff
@@ -43,9 +43,10 @@ When a linked/downstream app is pulled from its master/upstream app:
...
-- Corresponding forms in the master and linked app will have **different** unique ids.
-   - In order to keep these ids consistent from one pull to another, particularly when using multiple master apps (see below), part of the pull process maps the forms in the new linked app to the forms in the most recent version of the linked app. This overwrites the form ids copied from the master app with the form ids that are being used in the linked app. Any new forms (added to the master since the last time the linked app was pulled) will be assigned brand-new ids.
-   - Note on shadow forms: The mapping described above is based on XMLNS. Since a shadow form inherits its parent's XMLNS, shadow forms cannot be mapped. The result of this is that any shadow forms in the linked app are unstable: their unique ids change on **every** pull. This means that workflows that store a form's unique id, such as UCRs based on a form data source, will not work with shadow forms on linked apps.
+- Corresponding forms in the master and linked app will may have either the same or different unique ids, depending on how old the linked app is.
```
Comment (esoergel) [inline corehq/apps/linked_domain/README.md]:
```suggestion
- Corresponding forms in the master and linked app may have either the same or different unique ids, depending on how old the linked app is.
```

File: corehq/apps/linked_domain/README.md [hunk 2]
```diff
@@ -43,9 +43,10 @@ When a linked/downstream app is pulled from its master/upstream app:
...
+   - Older linked apps will have differing ids, while linked apps created after the deploy of [#25998](https://github.com/dimagi/commcare-hq/pull/25998) in December 2019 will use the same ids in both the linked and master apps.
+   - Linked apps that do not having form ids that match their master app have a mapping of master app form unique id => linked app form unique id, stored as [ResourceOverride](https://github.com/dimagi/commcare-hq/blob/15ceabdccf0ed49ed306462b3a154fe14886bf27/corehq/apps/app_manager/suite_xml/post_process/resources.py#L11) objects.
```
Comment (esoergel) [inline corehq/apps/linked_domain/README.md]:
```suggestion
   - Linked apps that do not have form ids that match their master app have a mapping of master app form unique id => linked app form unique id, stored as [ResourceOverride](https://github.com/dimagi/commcare-hq/blob/15ceabdccf0ed49ed306462b3a154fe14886bf27/corehq/apps/app_manager/suite_xml/post_process/resources.py#L11) objects.
```

**Review by esoergel — APPROVED** [review]
Is there any reasonable way to get rid of `ResourceOverride` at this point?

**Comment by orangejenny** [comment]
@esoergel The current `ResourceOverride` *data* can be dropped once all linked apps are on CommCare 2.47.4+. At that version and above, mobile can handle the xform resource IDs changing from one app version to the next. But the concept of resource overrides is likely to stick around, because this isn't the only time we've had a reason to manually fiddle with resource ids (multimedia versioning problems is the other big one).

**Comment by orangejenny** [comment]
(killed build)

**Pull request merged**